paasta_tools/tron_tools.py

```diff
@@ -884,6 +922,12 @@ def format_tron_action_dict(action_config: TronActionConfig):
     :param action_config: TronActionConfig
     """
     executor = action_config.get_executor()
+    # not building the Spark config in TronActionConfig constructor since it needs a KUBECONFIG file and
+    # fails in yelpsoa configs repo validation while getting the action configs.
+    if executor == "spark":
+        # build the complete Spark configuration
+        # TODO: add conditional check for Spark specific commands spark-submit, pyspark etc ?
+        action_config.action_spark_config = action_config.build_spark_config()
```

Comment:
can we instead use the `for_validation` flag? doing things this way will make interacting with these action configs in other scripts painful as folks will need to know that they need to manually call `build_spark_config()`